ukvm/ukvm_hv_freebsd_x86_64.c

```diff
@@ -201,11 +200,19 @@ void ukvm_hv_vcpu_loop(struct ukvm_hv *hv)
                         vme->u.inout.port);
 
             int nr = vme->u.inout.port - UKVM_HYPERCALL_PIO_BASE;
+            ukvm_gpa_t gpa = vme->u.inout.eax;
+
+            /* Guest has halted the CPU. */
+            if (nr == UKVM_HYPERCALL_HALT) {
```

Comment:
Please do the declaration and load of `gpa` at the start of this `if` block, and keep the original declaration and load of `gpa` below, before the call to `fn(...)`.

Reply:
changed